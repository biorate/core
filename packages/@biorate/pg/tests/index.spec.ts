import { expect } from 'chai';
import { root } from './__mocks__';

describe('@biorate/pg', function () {
  this.timeout(2e3);

  before(async () => {
    await root.$run();
    await root.connector!.current?.query(`DROP TABLE IF EXISTS test;`);
  });

  after(async () => {
    await root.connector!.current?.query(`DROP TABLE IF EXISTS test;`);
  });

  it('create table', async () =>
    await root.connector!.current?.query(
      `CREATE TABLE test (
         count int,
         text varchar(20)
      );`,
    ));

  it('insert rows into table', async () =>
    await root.connector!.current?.query(
      `INSERT INTO test (count, text) VALUES (1, 'test1'), (2, 'test2'), (3, 'test3');`,
    ));

  it('select rows from table', async () =>
    expect(
      (await root.connector!.current?.query(`SELECT * FROM test;`))!.rows,
    ).toMatchSnapshot());

  it('cursor', (done) => {
    const cursor = root.connector!.cursor('connection', `SELECT * FROM test;`);
    cursor.read(3, (err, rows) => {
      expect(err).toMatchSnapshot();
      expect(rows).toMatchSnapshot();
      cursor.close(() => done());
    });
  });

  it('stream', (done) => {
    const stream = root.connector!.stream('connection', `SELECT * FROM test;`);
    stream.on('data', (row) => expect(row).toMatchSnapshot());
    stream.on('end', () => done());
  });
});
