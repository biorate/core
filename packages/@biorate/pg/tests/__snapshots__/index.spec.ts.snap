// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@biorate/pg cursor 1`] = `null`;

exports[`@biorate/pg cursor 2`] = `
Array [
  Object {
    "count": 1,
    "text": "test1",
  },
  Object {
    "count": 2,
    "text": "test2",
  },
  Object {
    "count": 3,
    "text": "test3",
  },
]
`;

exports[`@biorate/pg select rows from table 1`] = `
Array [
  Object {
    "count": 1,
    "text": "test1",
  },
  Object {
    "count": 2,
    "text": "test2",
  },
  Object {
    "count": 3,
    "text": "test3",
  },
]
`;

exports[`@biorate/pg stream 1`] = `
Object {
  "count": 1,
  "text": "test1",
}
`;

exports[`@biorate/pg stream 2`] = `
Object {
  "count": 2,
  "text": "test2",
}
`;

exports[`@biorate/pg stream 3`] = `
Object {
  "count": 3,
  "text": "test3",
}
`;
